### PR TITLE
ceph-mds: Allow standby replay mds to be configured

### PR DESCRIPTION
**Pull request by secret104278**
Allow standby replay mds to be configured.
Resolve https://github.com/ceph/ceph-ansible/issues/5060

https://docs.ceph.com/docs/master/cephfs/standby/#configuring-standby-replay

Signed-off-by: Yu Chih secret104278@gmail.com

### CODE REVIEW

**Comment by dsavineau** [comment]
and run the generate_group_vars_sample.sh script to populate the group_vars samples.

It would be also great to rebase the PR

**Comment by dsavineau** [comment]
@batrick Is there any scenario where we don't want this setting set to true ? Should we ensure that we have at least 1 standby mds node available ? (ie: what happens if we are 3 mds and all are active)

File: roles/ceph-mds/tasks/create_mds_filesystems.yml [hunk 1]
```diff
@@ -77,3 +77,8 @@
   when:
     - mds_max_mds > 1
     - not rolling_update
+
+- name: set allow_standby_replay
+  command: "{{ container_exec_cmd | default('') }} ceph --cluster {{ cluster }} fs set {{ cephfs }} allow_standby_replay {{ mds_allow_standby_replay | bool | lower }}"
```
Comment (dsavineau) [inline roles/ceph-mds/tasks/create_mds_filesystems.yml]: You don't need the `| lower` filter

**Comment by batrick** [comment]
> @batrick Is there any scenario where we don't want this setting set to true ? Should we ensure that we have at least 1 standby mds node available ? (ie: what happens if we are 3 mds and all are active)

Ya, probably this should be off by default. User needs to be sure they have enough standbys to turn this on. I don't think ceph-ansible wants to get in the business of making that judgement.

**Comment by secret104278** [comment]
> > @batrick Is there any scenario where we don't want this setting set to true ? Should we ensure that we have at least 1 standby mds node available ? (ie: what happens if we are 3 mds and all are active)
> 
> Ya, probably this should be off by default. User needs to be sure they have enough standbys to turn this on. I don't think ceph-ansible wants to get in the business of making that judgement.

I have change the default value to false and add some warning that user should make sure they have enough mds to take over.

File: roles/ceph-mds/tasks/create_mds_filesystems.yml [hunk 2]
```diff
@@ -96,3 +96,8 @@
   when:
     - mds_max_mds > 1
     - not rolling_update
+
+- name: set allow_standby_replay
+  command: "{{ container_exec_cmd | default('') }} ceph --cluster {{ cluster }} fs set {{ cephfs }} allow_standby_replay {{ mds_allow_standby_replay | bool }}"
```
Comment (dsavineau) [inline roles/ceph-mds/tasks/create_mds_filesystems.yml]: you shouldn't use the bool filter here

**Comment by guits** [comment]
jenkins test centos-container-all_daemons

**Comment by guits** [comment]
jenkins test centos-container-all_in_one

**Comment by guits** [comment]
@secret104278  UP 🙂

**Pull request closed without merging**